core/templates/pages/story-editor-page/navbar/story-editor-navbar.component.spec.ts

```diff
@@ -106,20 +125,393 @@ describe('Story editor navbar directive', () => {
+      spyOnProperty(storyEditorStateService, 'onStoryInitialized')
+        .and.returnValue(mockStoryReinitializedEventEmitter);
+      spyOn(storyEditorStateService, 'getStoryWithUrlFragmentExists')
+        .and.returnValue(true);
+      spyOn(editableStoryBackendApiService, 'validateExplorationsAsync')
+        .and.returnValue(Promise.resolve([]));
+
+      component.ngOnInit();
+      fixture.detectChanges();
+
+      expect(component.explorationValidationIssues.length).toBe(0);
+
+      mockStoryReinitializedEventEmitter.emit();
+      tick();
+      fixture.detectChanges();
+
+      expect(component.explorationValidationIssues).toContain(
+        'Some chapters don\'t have exploration IDs provided.');
+    }));
+
+    it('should validate story without any validation errors ' +
+      'on initialization', () => {
+      story = storyObjectFactory.createFromBackendDict(storyBackendDict);
+      let mockStoryInitializedEventEmitter = new EventEmitter();
+
+      spyOn(storyEditorStateService, 'getStory').and.returnValue(story);
+      spyOnProperty(storyEditorStateService, 'onStoryInitialized')
+        .and.returnValue(mockStoryInitializedEventEmitter);
+
+      component.ngOnInit();
+      fixture.detectChanges();
+
+      expect(component.validationIssues.length).toBe(0);
+
+      mockStoryInitializedEventEmitter.emit();
+      fixture.detectChanges();
+
+      expect(component.validationIssues.length).toBe(0);
+    });
+
+    it('should validate story without any validation errors ' +
+      'on reinitalization', () => {
+      story = storyObjectFactory.createFromBackendDict(storyBackendDict);
+      let mockStoryReinitializedEventEmitter = new EventEmitter();
+
+      spyOnProperty(storyEditorStateService, 'onStoryReinitialized')
+        .and.returnValue(mockStoryReinitializedEventEmitter);
+      spyOn(storyEditorStateService, 'getStory').and.returnValue(story);
+
+      component.ngOnInit();
+      fixture.detectChanges();
+
+      expect(component.validationIssues.length).toBe(0);
+
+      mockStoryReinitializedEventEmitter.emit();
+      fixture.detectChanges();
+
+      expect(component.validationIssues.length).toBe(0);
+    });
+
+    it('should validate story without any validation errors ' +
+      'when undo operation is performed', () => {
+      story = storyObjectFactory.createFromBackendDict(storyBackendDict);
+      let mockUndoRedoChangeEventEmitter = new EventEmitter();
+
+      spyOn(storyEditorStateService, 'getStory').and.returnValue(story);
+      spyOn(undoRedoService, 'onUndoRedoChangeApplied$').and.returnValue(
+        mockUndoRedoChangeEventEmitter);
+
+      component.ngOnInit();
+      fixture.detectChanges();
+
+      expect(component.validationIssues.length).toBe(0);
+
+      mockUndoRedoChangeEventEmitter.emit();
+      fixture.detectChanges();
+
+      expect(component.validationIssues.length).toBe(0);
+    });
+  });
+
+  it('should unpublish story', fakeAsync(() => {
```

Comment:
We had the same scenario while testing the question-player-engine service, @seanlip suggested include both functions in the single test so it will act as pre-check and post-check to each other.